File: backend/models/ShoppingListModel/ShoppingList.cs

using FamilyPlanner.Models.FamilyModel;
using FamilyPlanner.Models.PlannedDayModel;
using FamilyPlanner.Models.PlannedTaskModel;
using FamilyPlanner.Models.UserModel;

namespace FamilyPlanner.Models.ShoppingListModel {
    public sealed class ShoppingList {
        public int Id { get; set; }
        public string Name { get; set; }
        public int? FamilyId { get; set; }
        public string? UserUid { get; set; }
        public List<ShoppingListItem>? Items { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int? PlannedDayId { get; set; }
        public PlannedDay? PlannedDay { get; set; }
    }
}

File: familyPlanner/src/models/shoppingList.ts
import { Family } from "./family";
import { KassalappProduct } from "./kassalappModel";
import { PlannedDay } from "./plannedDay";

export interface ShoppingList {
	id?: number;
	name: string;
	items?: ShoppingListItem[];
	familyId?: number;
	family?: Family;
	userUid?: string;
	dayKey?: string;
	startDate?: Date;
	endDate?: Date;
	plannedDayId?: number;
	plannedDay?: PlannedDay;
	completed?: boolean;
}

export interface ShoppingListItem {
	id?: number;
	name: string;
	quantity?: number;
	brand: string;
	ean?: string;
	url?: string;
	image?: string;
	description?: string;
	ingredients?: string;
	currentPrice: number;
	currentUnitPrice?: number;
	priceHistory?: PriceHistory[];
	allergens?: Allergens[];
	nutrition?: Nutrition[];
	createdAt?: Date;
	updatedAt?: Date;
	shoppingListId?: number;
	shoppingList?: ShoppingList;
}

interface PriceHistory {
	price?: number;
	date?: Date;
}

interface Allergens {
	code?: string;
	displayName?: string;
	contains?: string;
}

interface Nutrition {
	code?: string;
	displayName?: string;
	amount?: number;
	unit?: string;
}

export interface ShoppingListInput {}
